### DISCUSSION TITLE
List requried Scopes for PAT in "Action code" tab

### DISCUSSION THREAD

**Discussion by Andre601** [lowlighter/metrics#717] [Ideas]
First of all: This software/service is awersome. I like the GitHub-like style and the plugins it offers to customize it.
With that said do I see one big part that may stop people from using it and that is the fact of the possible requirement of specific scopes.

When I set some plugins using https://metrics.lecoq.io/ did I notice that the `Action code` doesn't provide a clear info on what scopes would be required for my PAT to make the action work without any issues.
Having said that is there also no real to know if one is even needed at all.

I therefore propose the idea of listing required Scopes in the `Action code` tab, depending on what plugins and configurations have been set. This could allow users to know straight away, if they can just make a basic PAT, or require some scopes to be checked.

Further would I quickly request some assistance in clearing this up right now, as I would like to include this into my own Profile README.

I currently have the following settings enabled/selected:

- **Template:** Classic
- **Base content**
  - Account sctivity
  - Community stats
  - Repositories metrics
  - Metadata
- **Additional plugins**
  - Isometric commit calendar
  - Most used languages
  - Recently starred repositories
  - People plugin
  - Follow-up of issues and pull requests

**Comment by lowlighter** [comment]
Hi!

Thanks for your feedback!

I'll add this this request to the backlog, it'd be indeed a nice addition 👍 
(It'll be tracked by #719)

With your current configuration, I don't think you need any scopes (you'll need `repo` one if you want to include private repositories though).

Usually the following scopes are required depending on what you want:
- `public_repo` scope for some plugins
- `read:org scope` for all organizations related metrics
- `repo` scope for all private repositories related metrics
  -  `read:user` scope may also be required for some private repositories related metrics

In the meantime, if you want to configure the action with additional plugins, you can get required scopes for each plugin known by checking their [respective documentation](https://github.com/lowlighter/metrics#-plugins).

Hope this helps!

**Reply by lowlighter** [reply]
This has been shipped since :+1:
